### PR TITLE
Add dir_name to ForceFieldMaker and **task_document_kwargs to from_ase_compatible_result

### PR DESCRIPTION
**Pull request by QuantumChemist**
I need to write the trajectory of a GAPrelax run into a file and then later access the directories of the files. That's why I added dir_name to ForceFieldRelaxMaker and **task_document_kwargs to ForceFieldTaskDocument.from_ase_compatible_result because as far as I can tell, that hasn't been implemented like that yet.

### CODE REVIEW

File: src/atomate2/forcefields/jobs.py [hunk 1]
```diff
@@ -74,6 +75,8 @@ def make(
                 "WARNING: A negative number of steps is not possible. "
                 "Behavior may vary..."
             )
+        if "dir_name" not in self.task_document_kwargs:
+            self.task_document_kwargs.update({"dir_name": os.getcwd()})
```
Comment (janosh) [inline src/atomate2/forcefields/jobs.py:79]: can be simplified to

```suggestion
        self.task_document_kwargs.setdefault("dir_name", os.getcwd())
```

**Review by janosh — CHANGES_REQUESTED** [review]
perhaps related? https://github.com/materialsproject/jobflow/pull/570

**Comment by QuantumChemist** [comment]
> perhaps related? [materialsproject/jobflow#570](https://github.com/materialsproject/jobflow/pull/570)

Having access to `job_dir` would be something handy in general! 
I went with `dir_name` like in other task documents.

**Comment by QuantumChemist** [comment]
Just wanted to ask if this PR is ok like that, @janosh ? :)

File: tests/forcefields/test_jobs.py [hunk 2]
```diff
@@ -252,3 +252,16 @@ def test_gap_relax_maker(si_structure: Structure, test_dir: Path, relax_cell: bo
...
+    # translate one atom to ensure a small number of relaxation steps are taken
+    si_structure.translate_sites(0, [0, 0, 0.1])
+
+    # generate job
+    job = CHGNetRelaxMaker(steps=25).make(si_structure)
+
+    # run the flow or job and ensure that it finished running successfully
+    responses = run_locally(job, ensure_success=True)
+
+    assert Path(responses[job.uuid][1].output.dir_name).exists()
```
Comment (janosh) [inline tests/forcefields/test_jobs.py:267]: i think we can move this line to one of the existing tests and avoid having to run another 25 relax steps in CI to keep runtime low

**Review by janosh — CHANGES_REQUESTED** [review]
apologies, didn't realize this was waiting on me. just 1 minor suggestions re testing

**Comment by QuantumChemist** [comment]
> apologies, didn't realize this was waiting on me. just 1 minor suggestions re testing

@janosh  no prob at all :) I adjusted the unit test now. I hope this is ok like that!

**Review by janosh — APPROVED** [review]
perfect, thank you!

**Comment by QuantumChemist** [comment]
@janosh can you please merge it? :) auto merge was disabled and I had to fix merge conflicts with the main branch!

**Pull request merged**